src/test/java/com/amplitude/AmplitudeTest.java

```diff
@@ -17,4 +40,99 @@ public void testGetInstance() {
     assertNotSame(d, e);
     assertNotSame(a, d);
   }
+
+  @ParameterizedTest
+  @ValueSource(booleans = {true, false})
+  public void testLogEventSuccess(boolean useBatch)
+      throws InterruptedException, NoSuchFieldException, IllegalAccessException,
+          AmplitudeInvalidAPIKeyException {
+    Amplitude amplitude = Amplitude.getInstance("test");
+    amplitude.init(apiKey);
+    amplitude.useBatchMode(useBatch);
+    amplitude.setLogMode(AmplitudeLog.LogMode.OFF);
+    List<Event> events = EventsGenerator.generateEvents(10, 5, 6);
+    HttpCall httpCall = getMockHttpCall(amplitude, useBatch);
+    Response response = new Response();
+    response.code = 200;
+    response.status = Status.SUCCESS;
+    CountDownLatch latch = new CountDownLatch(1);
+    when(httpCall.syncHttpCallWithEventsBuffer(anyList()))
+        .thenAnswer(
+            invocation -> {
+              latch.countDown();
+              return response;
+            });
+    for (Event event : events) {
+      amplitude.logEvent(event);
+    }
+    assertTrue(latch.await(1L, TimeUnit.SECONDS));
+    verify(httpCall, times(1)).syncHttpCallWithEventsBuffer(anyList());
+  }
+
+  @ParameterizedTest
+  @ValueSource(booleans = {true, false})
+  public void testLogEventWithInvalidKeyException(boolean useBatch)
+      throws InterruptedException, NoSuchFieldException, IllegalAccessException,
+          AmplitudeInvalidAPIKeyException {
+    Amplitude amplitude = Amplitude.getInstance("test");
+    amplitude.init(apiKey);
```

Comment:
Curious about this part, looks like all functions for testing the logEevent shared the same fake apiKey, why they will return different answer after call the `syncHttpCallWithEventsBuffer` ?

Reply:
The responses are mocked. The api key is just to complete the object?

Reply:
The api key here doesn't really matter, just leave it like how we normally use the library. Also we need the api key to complete the object and http call initiation.